### PR TITLE
Remove v1 code

### PR DESCRIPTION
**Pull request by EronWright**
### Proposed changes

Remove any now-obsolete v1 code and fix-up the Makefile.  Most of the top-level build targets work as expected, often by delegating to the sub-Makefiles.

TODO:

- [ ] CI scripts

Later:

- Deployment-related changes (to take place in a separate PR)
- Top-level tests

#### New Help Output
```
❯ make help

Usage:
  make <target>

General
  help             Display this help.

Development
  codegen          Generate CRDs and documentation
  generate-crdocs  Generate API Reference documentation into 'docs/crds/'.

Build
  build            Build the agent and operator binaries. 
  build-agent      Build the agent binary.
  build-operator   Build the operator manager binary.
  build-image      Build the operator image.
  push-image       Push the operator image.

Deployment
  install-crds     Install CRDs into the K8s cluster specified in ~/.kube/config.
  deploy           Deploy controller manager to the K8s cluster specified in ~/.kube/config.

Release
  prep             Prepare the next release.

Build Dependencies
  crdoc            Download crdoc locally if necessary. No version check.

```

### CODE REVIEW

File: deploy/crds/pulumi.com_programs.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.15.0
+    controller-gen.kubebuilder.io/version: v0.14.0
```
Comment (rquitales) [inline deploy/crds/pulumi.com_programs.yaml:6]: We should find a way to sync up our dev dependencies so we don't oscillate between versions.

File: deploy/helm/pulumi-operator/Chart.yaml [hunk 2]
```diff
@@ -9,7 +9,7 @@ icon: https://www.pulumi.com/logos/brand/avatar-on-white.svg
 
 type: application
 
-version: 0.8.0
+version: 0.9.0
```
Comment (rquitales) [inline deploy/helm/pulumi-operator/Chart.yaml:12]: Should we wait on bumping this until we want to trigger a helm release/publish?

File: docs/troubleshooting.md [hunk 3]
```diff
@@ -1,15 +1,13 @@
 # Troubleshooting
 
 * If a Stack is stuck due to conflicting updates from a previous failed run,
-you'll need to have the Pulumi program accessible locally, and manually cancel the stack.
+you'll need to manually unlock the stack. You can attach and use the stack's pod to run other commands.
 
   ```bash
-  pulumi stack select dev
+  kubectl exec --stdin --tty my-stack-0q4s6z9z
```
Comment (rquitales) [inline docs/troubleshooting.md:7]: Nice!

**Review by rquitales — APPROVED** [review]

**Review by blampe — APPROVED** [review]

File: deploy/crds/pulumi.com_programs.yaml (hunk 1 above)
Comment (EronWright) [reply deploy/crds/pulumi.com_programs.yaml:6]: Still, I agree that the makefile isn't doing enough here and will invesigate.

**Pull request merged**